src/tcgwars/logic/impl/gen4/PopSeries6.groovy

```diff
@@ -87,358 +92,89 @@ public enum PopSeries6 implements LogicCardInfo {
-            }
-          }
-          move "Ominous Wind", {
-            text "10 damage. Flip a coin. If heads, the Defending Pokémon is now Confused and can’t retreat during your opponent’s next turn."
-            energyCost P, C
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (DiamondPearl.DRIFLOON_46, this);
       case GIBLE_7:
         return basic (this, hp:HP050, type:COLORLESS, retreatCost:1) {
           weakness C, PLUS10
-          move "", {
-            text "Oran Berry. Remove 1 damage counter from Gible at the end of your turn."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
+          customAbility {
+            // Oran Berry: Remove 1 damage counter from Gible at the end of your turn.
+            delayedA {
+              before BEGIN_TURN,{
+                if (self.numberOfDamageCounters >= 1) {
+                  bc "Oran Berry activates"
+                  heal 10, self
+                }
+              }
             }
           }
           move "Surprise Attack", {
             text "30 damage. Flip a coin. If tails, this attack does nothing."
             energyCost C, C
             attackRequirement {}
             onAttack {
-              damage 0
+              flip { damage 30 }
             }
           }
-
         };
       case RIOLU_8:
-        return basic (this, hp:HP060, type:FIGHTING, retreatCost:1) {
-          weakness P, PLUS10
-          move "Wild Kick", {
-            text "30 damage. Flip a coin. If tails, this attack does nothing."
-            energyCost F
-            attackRequirement {}
-            onAttack {
-              damage 0
-            }
-          }
-
-        };
+        return copy (DiamondPearl.RIOLU_61, this);
       case PIKACHU_9:
         return basic (this, hp:HP050, type:LIGHTNING, retreatCost:1) {
           weakness F, PLUS10
           resistance M, MINUS20
-          move "", {
-            text "Oran Berry. Remove 1 damage counter from Pikachu at the end of your turn."
-            energyCost ()
-            attackRequirement {}
-            onAttack {
-              damage 0
+          customAbility {
+            // Oran Berry: Remove 1 damage counter from Gible at the end of your turn.
+            delayedA {
+              before BEGIN_TURN,{
+                if (self.numberOfDamageCounters >= 1) {
+                  bc "Oran Berry activates"
+                  heal 10, self
+                }
+              }
             }
           }
           move "Spark", {
             text "10 damage. Does 10 damage to 2 of your opponent’s Benched Pokémon."
             energyCost L, C
             attackRequirement {}
             onAttack {
-              damage 0
+              damage 10
+              if (opp.bench) {
+                multiSelect(opp.bench, 2, text).each {
+                  targeted(it) {
```

Comment:
targeted is not appropriate here. It will cause abilities that prevent all effects of attacks except damage to block the damage.